### PR TITLE
Initial import of the OpenTracing testbed.

### PR DESCRIPTION
**Pull request by carlosalberto**
Originated from #408 but had to be split to merge the `InMemoryTracer` first ;)

Observe I had to make `InMemoryTracer` public as it was required by the `testbed` and its cases, which are all in their respective submodules.

In any case, there's already an issue for using the SDK instead of the `InMemoryTracer` in the near future.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@carlosalberto I know that you will hate me now :) can you use the InMemory from the SDK? See the tests for InMemory how I configured and used it.

**Comment by carlosalberto** [comment]
Heh, oh well. Just right in time, I think ;) will update my PR tomorrow then, and then we can finally merge these tests.

**Comment by bogdandrutu** [comment]
Status of this?

**Comment by carlosalberto** [comment]
There's a small 'bug' in the SDK which was preventing my latest updates to work - just tracked it down. Will cook a PR for that one, and will iterate on this PR afterwards (which should be today).

File: build.gradle [hunk 1]
```diff
@@ -119,6 +119,9 @@ subprojects {
 
                 // Compatibility layer
                 opentracing: "io.opentracing:opentracing-api:${opentracingVersion}",
+                slf4jsimple: 'org.slf4j:slf4j-simple:1.7.25', // Testing
```
Comment (bogdandrutu) [inline build.gradle]: Move these 10 lines bellow where "Test dependencies" are.

**Comment by carlosalberto** [comment]
Done - there's still a small issue/TODO regarding https://github.com/open-telemetry/opentelemetry-specification/issues/67 - we can address/update this case later on ;)

**Comment by bogdandrutu** [comment]
Some tests seem to fail, I think I fixed the travis config to also fail https://github.com/open-telemetry/opentelemetry-java/pull/463 (or maybe I am wrong) :)

**Comment by carlosalberto** [comment]
Heh, thanks for the note. Interestingly enough I can't reproduce it locally but I think I got an idea where/why it's failing.

**Comment by carlosalberto** [comment]
@bogdandrutu Agreed. This test relies too much on this (and it used to rely even more, hah), but I never had time to update the examples contributed by the users. But that's an item to go into my TODO.

**Pull request merged**